Client/game_sa/CWorldSA.cpp

```diff
@@ -350,6 +388,157 @@ bool CWorldSA::ProcessLineOfSight(const CVector* vecStart, const CVector* vecEnd
             }
         }
     }
+
+    if (outMatInfo && targetEntity && targetEntity->m_pRwObject) {
+        struct Context {
+            float               minHitDistSq{};         //< [squared] hit distance from the line segment's origin
+            CVector             originOS, endOS, dirOS; //< Line origin, end and dir [in object space]
+            CMatrix             entMat, entInvMat;      //< The hit entity's matrix, and it's inverse
+            RpTriangle*         hitTri{};               //< The triangle hit
+            RpAtomic*           hitAtomic{};            //< The atomic of the hit triangle's geometry
+            RpGeometry*         hitGeo{};               //< The geometry of the hit triangle
+            CVector             hitBary{};              //< Barycentric coordinates [on the hit triangle] of the hit 
+            CVector             hitPosOS{};             //< Hit position in object space
+            CEntitySAInterface* entity{};               //< The hit entity
+        } c = {};
+
+        c.entity = targetEntity;
+
+        // Get matrix, and it's inverse
+        targetEntity->Placeable.matrix->ConvertToMatrix(c.entMat);
+        c.entInvMat = c.entMat.Inverse();
+
+        // ...to transform the line origin and end into object space
+        c.originOS     = c.entInvMat.TransformVector(*vecStart);
+        c.endOS        = c.entInvMat.TransformVector(*vecEnd);
+        c.dirOS        = c.endOS - c.originOS;
+        c.minHitDistSq = c.dirOS.LengthSquared(); // By setting it to this value we avoid collisions that would be detected after the line segment [but are still ont the ray]
+
+        //const auto lineMagSq = (ctx.originOS - ctx.endOS).LengthSquared();
+
+        // Do raycast against the DFF to get hit position material UV and name
+        // This is very slow
+        // Perhaps we could paralellize it somehow? [OpenMP?]
+        const auto ProcessOneAtomic = [](RpAtomic* a, void* data) {
```

Comment:
Please format the entire block properly.

Reply:
Should I just run clang-format on it?